ruby_on_rails/advanced_forms_and_activerecord/installing_psql.md

```diff
@@ -0,0 +1,239 @@
+
+Instead, we will set up our own user to avoid switching to `postgres` all the time.
+
+<div class="lesson-note" markdown="1">
+#### IMPORTANT
+You'll want to take note of your Linux username so we can create the right role! The terminal prompt, by default, is configured in this format:
+
+~~~bash
+<linux_username>@<machine_name>: 
+~~~
+</div>
+
+#### 3.2 Creating a New Role
+With all your information ready, let's create a role in Postgre! The command to do so is:
+
+~~~bash
+sudo -u postgres createuser --interactive
+~~~
+
+Make sure that you enter your Linux user name as the name of the role, and be sure to make that user a superuser. Setting up a role this way means that we leverage "peer authentication" which makes using the local database very easy from Rails.
+
+#### 3.3 Creating the Role Database
+One other important step in setting up Postgre is that each role must have its own database of the same name. Without it, the role we just created will not be able to log in or interact with the Postgre server.
+
+You can try to run `psql` now, but you will get an error that the database does not exist. Not to worry, let's create one to resolve fix this:
+
+~~~bash
+sudo -u postgres createdb <linux_username>
+~~~
+
+Now our role is fully set up: we've got `<role_name>` and that role has a database.
+
+#### 3.4 Securing Our New Role
+One important thing that we have to do is to set up a password for our new role so that the data is protected. Now that our role is set up, we can actually use it to administer the Postgre server. All you have to do is enter this command to get into the PostgreSQL prompt:
+
+~~~bash
+psql
+~~~
+
+You should see the PostgreSQL prompt come up with the new role we just created, like so:
+
+~~~
+<role_name>=#
+~~~
+
+If you don't see a similar prompt, then reach out on Discord for some help. 
+
+If you **do** see a similar prompt, then we can create a password for the role like so:
+
+~~~
+\password <role_name>
+~~~
+
+You'll be prompted to enter a password and to verify it. Once you are done, the prompt will return to normal. Now, we will configure the permissions for our new role:
+
+~~~
+grant all privileges on database <role_database_name> to <role_name>;
+~~~
+
+Remember that you should change the `<role_database_name>` and `<role_name>` (they should both the same)! If you see `GRANT` in response to the command, then you can type `\q` to exit the prompt.
+
+#### 3.5 Saving Access Information in the Environment
+After finishing our configuration, the last step is save it into the environment to access later.
+
+In order to save our password to the environment, we can run this command:
+
+~~~bash
+echo 'export DATABASE_PASSWORD="<role_password>"' >> ~/.bashrc
+~~~
+
+Note here the name we've chosen for our environment variable: `DATABASE_PASSWORD`. Also, remember to update `<role_password>` in the command to what was set above!
```

Comment:
I'm personally a little confused by implications of this - is setting up one database and password for it enough for any Rails project I do or do I have to hold multiple of these?

Reply:
Excellent point. The role and password combination allows us to access the server to do database operations like create, drop or edit databases.

I'll add a note about this in the section previous section.